rskj-core/src/main/java/co/rsk/trie/TrieImpl.java

```diff
@@ -1030,11 +1028,10 @@ public Trie getSnapshotTo(Keccak256 hash) {
         Trie newTrie = this.store.retrieve(hash.getBytes());
 
         if (newTrie == null) {
-            String strHash = hash.toHexString();
-            logger.error(ERROR_NON_EXISTENT_TRIE_LOGGER, strHash);
-            panicProcessor.panic(PANIC_TOPIC, ERROR_CREATING_TRIE + " " + strHash);
-
-            throw new TrieSerializationException(ERROR_CREATING_TRIE + " " + strHash, null);
+            String log = String.format(ERROR_NON_EXISTENT_TRIE, hash.toHexString());
+            logger.error(log);
+            panicProcessor.panic(PANIC_TOPIC, log);
+            throw new IllegalStateException(log);
```

Comment:
I would use `IllegalArgumentException` here given that exception is a direct outcome of a flawed argument